### ISSUE TITLE
[Bug]: Top toolbar lacking semantics

### ISSUE THREAD

**Issue by sara-temby-design** [storybookjs/storybook#25918]
### Describe the bug

The HTML markup for the top toolbar needs semantics. The buttons should be marked up as an unordered list to indicate the number of items clearly, and ideally, this would be grouped into a labeled `<section>`. 
![bug1](https://github.com/storybookjs/storybook/assets/76465110/d79383b3-aa50-440d-b7d0-5b6ccf5f3a83)

### To Reproduce

Inspect top toolbar elements.

**Comment by sara-temby-design** [comment]
#A11y issue

**Comment by mehm8128** [comment]
I'll work for this issue.

**Comment by Sidnioulz** [comment]
Still left to do:
* Switch the `span` used as an `aria-labelledby` to a heading?
* Make sure children are part of an unordered list, so the number of tools is announced early, by wrapping each child item inside a `li` and the whole content inside a `ul`

Pending verification, as I'm unsure whether this wouldn't cause UX regressions for non-blind users; I've asked around and am waiting for feedback from more experienced people:
* Implement `toolbar` role and add orientation ARIA attribute
* Implement [keyboard navigation pattern](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/toolbar_role) required for toolbars

**Comment by Sidnioulz** [comment]
I've had confirmation of user research with the toolbar role and keyboard navigation in a variety of contexts, including VoiceOver users, with positive results. Test participants did initially tab out instead of arrowing but did not mention that as a blocker, so they understood they needed to use arrow instead from the announced role. We should be going ahead with `toolbar`.

We'll probably want to wrap that `toolbar` in a `region` landmark with its own heading for better discoverability.
